Extensions/Polygon.cpp

```diff
@@ -0,0 +1,187 @@
+//--------------------------------------------------------------------------------
+//constexpr float PixelAlphaGain = 255.0;
+constexpr float LoAlphaThreshold = 1.0 / 32.0;
+constexpr float HiAlphaThreshold = 1.0 - LoAlphaThreshold;
+
+//--------------------------------------------------------------------------------
+// Calculates closest distance of point to a vector (with origin at 0,0)
+float TFT_eSPI::calcDistance(tFPoint *aPoint, tFPoint *aVector) {
+    float lH = fmaxf(fminf((aPoint->X * aVector->X + aPoint->Y * aVector->Y) / (aVector->X * aVector->X + aVector->Y * aVector->Y), 1.0f), 0.0f);
+    float lDx = aPoint->X - (aVector->X * lH);
+    float lDy = aPoint->Y - (aVector->Y * lH);
+    return sqrtf(lDx * lDx + lDy * lDy);
+}
+
+//--------------------------------------------------------------------------------
+// Calculates closest distance of point to a vector (with origin at 0,0)
+// Result
+// 0: No intersection
+// 1: Intersection with ascending vector 
+// -1: Intersection with descending vector 
+int TFT_eSPI::checkIntersection(tFPoint *aPoint, tFPoint *aVector) {
```

Comment:
As you are not mutating the arguments consider passing them by value or const reference.

Reply:
Hi B3rn475,
isn't a * parameter as good as a "pass by reference"?

Reply:
Se other comment.